google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -167,50 +168,42 @@ private boolean isManagedLibrariesEnabled() {
 
     @Override
     public void afterLibraryAdded(final Library addedLibrary) {
-      AppEngineStandardMavenLibrary library = AppEngineStandardMavenLibrary
-          .getLibraryByMavenDisplayName(addedLibrary.getName());
+      Optional<AppEngineStandardMavenLibrary> libraryOptional =
+          AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName());
 
-      if (library == null) {
-        return;
-      }
+      libraryOptional.ifPresent(
+          library -> {
+            final DependencyScope scope = library.getScope();
 
-      final DependencyScope scope = library.getScope();
+            ApplicationManager.getApplication().runWriteAction(() -> {
```

Comment:
Just to be safe, did you test this out in the UI to make sure runWriteAction does the same thing as before?